tests/src/Functional/OgComplexWidgetTest.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\og\Functional\OgComplexWidgetTest.
+ */
+
+namespace Drupal\Tests\og\Functional;
+
+use Drupal\Core\Entity\Display\EntityFormDisplayInterface;
+use Drupal\Core\Entity\Query\QueryInterface;
+use Drupal\node\Entity\Node;
+use Drupal\node\NodeInterface;
+use Drupal\og\Og;
+use Drupal\og\OgGroupAudienceHelper;
+use Drupal\og\Plugin\Field\FieldType\OgMembershipReferenceItemList;
+use Drupal\simpletest\BrowserTestBase;
+use Drupal\simpletest\ContentTypeCreationTrait;
+use Drupal\simpletest\NodeCreationTrait;
+
+/**
+ * Tests the complex widget.
+ *
+ * @group og
+ */
+class OgComplexWidgetTest extends BrowserTestBase {
+
+  use ContentTypeCreationTrait;
+  use NodeCreationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = ['node', 'og'];
+
+  /**
+   * {@inheritdoc}
+   */
+  function setUp() {
+    parent::setUp();
+
+    // Create a "group" node type and turn it into a group type.
+    $this->createContentType(['type' => 'group']);
+    Og::groupManager()->addGroup('node', 'group');
+
+    // Add a group audience field to the "post" node type, turning it into a
+    // group content type.
+    $this->createContentType(['type' => 'post']);
+    Og::createField(OgGroupAudienceHelper::DEFAULT_FIELD, 'node', 'post');
+
+    // Make the group audience field visible in the default form display.
+    // @todo Remove this once issue #144 is in.
+    // @see https://github.com/amitaibu/og/issues/144
+    /** @var EntityFormDisplayInterface $form_display */
+    $form_display = \Drupal::entityTypeManager()->getStorage('entity_form_display')->load('node.post.default');
+    $widget = $form_display->getComponent('og_group_ref');
```

Comment:
as indicated in the issue, this one troubles me a bit (even if it's an intermediate step). I'd be happy to see the next step to make this work for all audience fields